src/Raven.Client/ServerWide/Operations/DatabaseRecordBuilder.cs

```diff
@@ -0,0 +1,775 @@
+        _databaseRecord.Sharding.Orchestrator.Topology = topology;
+        return this;
+    }
+
+    IShardedTopologyConfigurationBuilder IShardedTopologyConfigurationBuilder.Orchestrator(Action<IOrchestratorTopologyConfigurationBuilder> builder)
+    {
+        if (builder == null)
+            throw new ArgumentNullException(nameof(builder));
+
+        builder(this);
+        return this;
+    }
+
+    IShardedTopologyConfigurationBuilder IShardedTopologyConfigurationBuilder.AddShard(int shardNumber, DatabaseTopology topology)
+    {
+        if (topology == null)
+            throw new ArgumentNullException(nameof(topology));
+
+        _databaseRecord.Sharding.Shards ??= new Dictionary<int, DatabaseTopology>();
+        _databaseRecord.Sharding.Shards.Add(shardNumber, topology);
+        return this;
+    }
+
+    IShardedTopologyConfigurationBuilder IShardedTopologyConfigurationBuilder.AddShard(int shardNumber, Action<IShardTopologyConfigurationBuilder> builder)
+    {
+        if (builder == null)
+            throw new ArgumentNullException(nameof(builder));
+
+        _shardTopology = new DatabaseTopology();
+        try
+        {
+            builder(this);
+
+            _databaseRecord.Sharding.Shards ??= new Dictionary<int, DatabaseTopology>();
+            _databaseRecord.Sharding.Shards.Add(shardNumber, _shardTopology);
+        }
+        finally
+        {
+            _shardTopology = null;
+        }
+
+        return this;
+    }
+
+    IShardTopologyConfigurationBuilder ITopologyConfigurationBuilderBase<IShardTopologyConfigurationBuilder>.AddNode(string nodeTag)
+    {
+        if (string.IsNullOrEmpty(nodeTag))
+            throw new ArgumentException("Value cannot be null or empty.", nameof(nodeTag));
+
+        _shardTopology.Members.Add(nodeTag);
+        return this;
+    }
+
+    ITopologyConfigurationBuilder ITopologyConfigurationBuilderBase<ITopologyConfigurationBuilder>.AddNode(string nodeTag)
+    {
+        _databaseRecord.Topology ??= new DatabaseTopology();
+        _databaseRecord.Topology.Members.Add(nodeTag);
+
+        return this;
+    }
+
+    IDatabaseRecordBuilderBase IDatabaseRecordBuilder.WithTopology(DatabaseTopology topology)
+    {
+        _databaseRecord.Topology = topology ?? throw new ArgumentNullException(nameof(topology));
+        return this;
+    }
+
+    IDatabaseRecordBuilderBase IDatabaseRecordBuilder.WithTopology(Action<ITopologyConfigurationBuilder> builder)
+    {
+        if (builder == null)
+            throw new ArgumentNullException(nameof(builder));
+
+        builder(this);
+        return this;
+    }
+
+    IDatabaseRecordBuilderBase IDatabaseRecordBuilder.WithReplicationFactor(int replicationFactor)
+    {
+        _databaseRecord.Topology ??= new DatabaseTopology();
+        _databaseRecord.Topology.ReplicationFactor = replicationFactor;
+
+        return this;
+    }
+
+    private void HandleDisabled()
```

Comment:
There is only one caller of this method. Can we move `_databaseRecord.Disabled = true;` directly there? The same applies to other `HandleXXX()` methods